docs/versioned_docs/version-v0.85.0-alpha-testnet.0/glossary.md

```diff
@@ -3,44 +3,80 @@ title: Glossary
 tags: [protocol, glossary]
 ---
 
+import { Glossary } from '@site/src/components/Snippets/glossary_snippets';
+
+### Aztec
+
+<Glossary.Tools.aztec />
+
+### Aztec Wallet
+
+<Glossary.Tools.aztec_wallet />
+
 ### `aztec-nargo`
 
-The command line tool used to compile Aztec contracts. It gets its name from the Noir compiler, `nargo`. `aztec-nargo` is really just a specific version of `nargo`. You can read more about `nargo` [here](#nargo).
+<Glossary.Tools.aztec_nargo />
+
+You can read more about `nargo` [here](#nargo).
+
+### `aztec-up`
+
+<Glossary.Tools.aztec_up />
+
+### Aztec.js
+
+<Glossary.Libs.aztec_js />
+
+Read more and review the source code [here](https://github.com/AztecProtocol/aztec-packages/blob/#include_aztec_version/yarn-project/aztec.js).
 
 ### Aztec.nr
 
-A [Noir](https://noir-lang.org) framework for writing smart contracts on Aztec.
+<Glossary.Libs.aztec_nr />
 
 Read more and review the source code [here](https://aztec.nr).
 
 ### Barretenberg
 
-Aztec's cryptography back-end. Refer to the graphic at the top of [this page](https://medium.com/aztec-protocol/explaining-the-network-in-aztec-network-166862b3ef7d) to see how it fits in the Aztec architecture.
+<Glossary.Barretenberg />
+
+### bb / bb.js
 
-Barretenberg's source code can be found [here](https://github.com/AztecProtocol/barretenberg).
+`bb` (CLI) and its corresponding `bb.js` (node module) are tools that prove and verify circuits. It also has helpful functions such as: writing solidity verifier contracts, checking a witness, and viewing a circuit's gate count.
 
 ### `nargo`
 
-With `nargo`, you can start new projects, compile, execute, prove, verify, test, generate solidity contracts, and do pretty much all that is available in Noir.
+With `nargo`, you can start new projects, compile, execute, and test your Noir programs.
 
 You can find more information in the nargo installation docs [here](https://noir-lang.org/docs/getting_started/installation/) and the nargo command reference [here](https://noir-lang.org/docs/reference/nargo_commands).
 
 ### Noir
 
-Noir is a Domain Specific Language for SNARK proving systems. It is used for writing smart contracts in Aztec because private functions on Aztec are implemented as SNARKs to support privacy-preserving operations.
+<Glossary.Noir />
+
+### Noir Language Server
+
+The Noir Language Server can be used in vscode to facilitate writing programs in Noir. The Noir LSP addon allows the dev to choose their tool, nargo or aztec-nargo, when writing a pure Noir program or an Aztec smart conract.
```

Comment:
```suggestion
The Noir Language Server can be used in vscode to facilitate writing programs in Noir by providing syntax highlighting, circuit introspection and an execution interface. The Noir LSP addon allows the dev to choose their tool, nargo or aztec-nargo, when writing a pure Noir program or an Aztec smart conract.
```

Reply:
I was thinking of expanding definitions to different teirs almost. From just this simple context tier with quick definitions, then adding another tier with deeper explanations.
The location wishing to use the snippet could then choose if it wants the shorter verser the longer. The glossary mostly had short definitions, and that also suits the first location of the snippets.

I'll be adding this and other glossary fixes to all versions in this PR.